File: swazy-ui/src/app/models/booking.ts
export interface Booking {
  id: string;
  confirmationCode: string;
  customerName: string;
  customerEmail: string;
  customerPhone: string;
  serviceName: string;
  employeeName: string;
  bookingDate: Date;
  notes?: string;
}

export interface BookingDetailsResponse {
  id: string;
  confirmationCode: string;
  startTime: string;
  endTime: string;
  serviceName: string;
  serviceDuration: number;
  servicePrice: number;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  notes?: string;
  employeeName?: string;
  businessName: string;
  businessAddress?: string;
  businessPhone?: string;
  employeeId?: string;
  bookedByUserId?: string;
}

File: Api.Swazy/Api.Swazy/Models/Entities/Booking.cs
using Api.Swazy.Models.Base;

namespace Api.Swazy.Models.Entities;

public class Booking : BaseEntity
{
    public string ConfirmationCode { get; set; } = string.Empty;
    public DateTimeOffset BookingDate { get; set; }
    public string? Notes { get; set; } = string.Empty;
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;

    public Guid BusinessServiceId { get; set; }
    public Guid? EmployeeId { get; set; }
    public Guid? BookedByUserId { get; set; }

    public virtual BusinessService BusinessService { get; set; }
    public virtual User? Employee { get; set; }
    public virtual User? BookedByUser { get; set; }
}
